OpenRA.Mods.Common/UpdateRules/Rules/20231010/ReplaceCloakPalette.cs

```diff
@@ -0,0 +1,49 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ReplaceCloakPalette : UpdateRule
+	{
+		public override string Name => "Change default Cloak style from Palette to Alpha.";
+
+		public override string Description =>
+			"Cloak has gained several new rendering modes\n" +
+			"and its default behaviour has changed from using a palette to native alpha.";
+
+		readonly List<string> locations = new();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Count > 0)
+				yield return "Cloak no longer defaults to replacing the actor's palette.\n" +
+					"If you wish to keep the previous behavior you wish to change the\n" +
+					"Cloak definitions on the following actor definitions:\n" +
+					UpdateUtils.FormatMessageList(locations);
+
+			locations.Clear();
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNodeBuilder actorNode)
+		{
+			foreach (var cloak in actorNode.ChildrenMatching("Cloak"))
+			{
+				cloak.RemoveNodes("Palette");
+				cloak.RemoveNodes("IsPlayerPalette");
+				locations.Add($"{actorNode.Key} ({actorNode.Location.Filename})");
```

Comment:
```suggestion
				var palette = cloak.LastChildMatching("Palette");
				if (palette != null)
				{
					if (string.IsNullOrEmpty(palette.Value.Value))
					{
						cloak.RemoveNode(palette);
						cloak.AddNode("CloakStyle", "None");
					}
					else
					{
						palette.RenameKey("CloakedPalette");
						cloak.AddNode("CloakStyle", "Palette");
					}
				}
				else
				{
					cloak.AddNode("CloakedPalette", "cloak");
					cloak.AddNode("CloakStyle", "Palette");
				}
```

If it's fairly easy to add a proper rule then why not add it?

Reply:
Because its not that easy!

The suggested changes will switch RA's missile submarine to use the phase transport cloak effect.

Reply:
it would keep the submerged palette?

Reply:
The submerged palette is defined on ^Submarine, which is updated as you expect. `MSUB` then overrides the cloak delay, which the rule sees as not having a Palette key so it incorrectly adds `CloakedPalette: cloak`.

Reply:
ah right, always the same problem when dealing with defaults.